airgen/src/lib.rs

```diff
@@ -295,11 +309,128 @@ impl<'a> ASMPILConverter<'a> {
                         id: d.operation.id.id.clone(),
                         params: d.operation.params.clone(),
                     },
-                    selector_idx,
+                    // this will be set later, after compatible links are merged
+                    selector_idx: None,
                 })
                 .unwrap()
                 .clone(),
             is_permutation,
         }
     }
+
+    /// Process each link and then combine compatible links.
+    /// Links can be merged iff:
+    /// - they originate from the same machine instance
+    /// - they target the same instance.operation
+    /// - they are of the same kind (permutation/lookup)
+    /// - their flags are mutually exclusive
+    /// Right now we only consider links from different instructions,
+    /// as a single instruction can be active at a time
```

Comment:
Could you maybe (can also be inside the function) explain what merging links does (with an example)?